include/nbl/asset/filters/kernels/WeightFunctions.h

```diff
@@ -0,0 +1,308 @@
+#ifndef _NBL_ASSET_MATH_FUNCTIONS_H_INCLUDED_
+#define _NBL_ASSET_MATH_FUNCTIONS_H_INCLUDED_
+
+#include <limits>
+#include <cstdint>
+
+namespace nbl::asset
+{
+
+struct SDiracFunction
+{
+	// We can use std::nextafter after C++23
+	constexpr static inline float min_support = -1e-6f;
+	constexpr static inline float max_support = +1e-6f;
+	constexpr static inline uint32_t k_smoothness = 0;
+
+	template<int32_t derivative=0>
+	static inline double weight(float x)
+	{
+		if (x != 0.f)
+			return 0.0;
+
+		if constexpr (derivative == 0)
+		{
+			return std::numeric_limits<double>::infinity();
+		}
+		else
+		{
+			static_assert(false);
+			return core::nan<double>();
+		}
+	}
+};
+
+// Standard Box function, symmetric, value in the support is 1, integral is 1, so support must be [-1/2,1/2)
+struct SBoxFunction
+{
+	constexpr static inline float min_support = -0.5f;
+	constexpr static inline float max_support = +0.5f;
```

Comment:
ok, if you're not making the box filter slightly large here, then you need to apply a slight stretch in `CBlitUtilities` when you detect it being used

https://discord.com/channels/593902898015109131/1043939034776473710/1062733502266744842

Something like `1.f+1.f/8192.f` stretch

Reply:
ACtually not important if the default is double convolution.